File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row = set()
        col = set()
        square = set()

        for i in range(9):
            for j in range(9):
                if board[i][j] != '.':
                    if (((i, board[i][j]) in row) or
                        ((j, board[i][j]) in col) or
                        ((i // 3, j // 3, board[i][j]) in square)):
                        return False
                    row.add((i, board[i][j]))
                    col.add((j, board[i][j]))
                    square.add((i // 3, j // 3, board[i][j]))

        return True

File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Set<String> row = new HashSet();
        Set<String> col = new HashSet();
        Set<String> square = new HashSet();

        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                if (board[i][j] != '.') {
                    String rowChk = Integer.toString(i) + board[i][j];
                    String colChk = Integer.toString(j) + board[i][j];
                    String sqChk = Integer.toString(i / 3) + Integer.toString(j / 3) + board[i][j];
                    if (row.contains(rowChk) || col.contains(colChk) || square.contains(sqChk)) {
                        return false;
                    }
                    row.add(rowChk);
                    col.add(colChk);
                    square.add(sqChk);
                }
            }
        }

        return true;
    }
}